### ISSUE TITLE
Building Caddy from source fails when importing one of the DNS provider plugins

### ISSUE THREAD

**Issue by whalehub** [caddyserver/dnsproviders#75]
I'm trying to build Caddy from source with Go v1.12.6 by first running `go get github.com/caddyserver/caddy` and then running `go build` inside a folder with the following `caddy.go` and `go.mod` files:

```
package main

import (
        "github.com/caddyserver/caddy/caddy/caddymain"

        _ "github.com/caddyserver/dnsproviders/gandiv5"
)

func main() {
	caddymain.EnableTelemetry = false
	caddymain.Run()
}
```
```
module caddy

go 1.12
```
```
root@debian:~# go get github.com/caddyserver/caddy
go: finding github.com/caddyserver/caddy v1.0.1
go: finding github.com/jimstudt/http-authentication v0.0.0-20140401203705-3eca13d6893a
go: finding github.com/hashicorp/go-syslog v1.0.0
go: finding github.com/klauspost/cpuid v1.2.0
go: finding github.com/kylelemons/godebug v0.0.0-20170820004349-d65d576e9348
go: finding github.com/naoina/toml v0.1.1
go: finding github.com/naoina/go-stringutil v0.1.0
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/lucas-clemente/quic-go v0.10.2
go: finding golang.org/x/net v0.0.0-20190328230028-74de082e2cca
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568
go: finding github.com/go-acme/lego v2.5.0+incompatible
go: finding github.com/google/uuid v1.1.1
go: finding golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
go: finding gopkg.in/natefinch/lumberjack.v2 v2.0.0
go: finding gopkg.in/mcuadros/go-syslog.v2 v2.2.1
go: finding github.com/mholt/certmagic v0.6.2-0.20190624175158-6a42ef9fe8c2
go: finding golang.org/x/text v0.3.0
go: finding github.com/lucas-clemente/aes12 v0.0.0-20171027163421-cd47fb39b79f
go: finding golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
go: finding github.com/BurntSushi/toml v0.3.1
go: finding github.com/lucas-clemente/quic-go-certificates v0.0.0-20160823095156-d2f86524cced
go: finding github.com/bifurcation/mint v0.0.0-20180715133206-93c51c6ce115
go: finding github.com/golang/mock v1.2.0
go: finding github.com/russross/blackfriday v0.0.0-20170610170232-067529f716f4
go: finding github.com/onsi/gomega v1.5.0
go: finding golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a
go: finding golang.org/x/net v0.0.0-20190125091013-d26f9f9a57f3
go: finding gopkg.in/square/go-jose.v2 v2.2.2
go: finding golang.org/x/net v0.0.0-20180906233101-161cd47e91fd
go: finding gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405
go: finding github.com/hashicorp/golang-lru v0.0.0-20180201235237-0fb14efe8c47
go: finding golang.org/x/crypto v0.0.0-20190123085648-057139ce5d2b
go: finding github.com/onsi/ginkgo v1.8.0
go: finding golang.org/x/sys v0.0.0-20190124100055-b90733256f2e
go: finding github.com/lucas-clemente/quic-clients v0.1.0
go: finding github.com/stretchr/testify v1.3.0
go: finding golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
go: finding github.com/marten-seemann/qtls v0.2.3
go: finding golang.org/x/sys v0.0.0-20180909124046-d0be0721c37e
go: finding github.com/onsi/ginkgo v1.6.0
go: finding github.com/fsnotify/fsnotify v1.4.7
go: finding gopkg.in/yaml.v2 v2.2.1
go: finding gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: finding github.com/golang/protobuf v1.2.0
go: finding github.com/miekg/dns v1.1.3
go: finding gopkg.in/fsnotify.v1 v1.4.7
go: finding github.com/cenkalti/backoff v2.1.1+incompatible
go: finding github.com/hpcloud/tail v1.0.0
go: finding github.com/cheekybits/genny v0.0.0-20170328200008-9127e812e1e9
go: finding github.com/stretchr/objx v0.1.0
go: finding golang.org/x/sys v0.0.0-20190228124157-a34e9553db1e
go: finding golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/davecgh/go-spew v1.1.0
go: finding github.com/gorilla/websocket v1.4.0
go: downloading github.com/caddyserver/caddy v1.0.1
go: extracting github.com/caddyserver/caddy v1.0.1
go: downloading github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568
go: downloading github.com/google/uuid v1.1.1
go: downloading github.com/mholt/certmagic v0.6.2-0.20190624175158-6a42ef9fe8c2
go: extracting github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568
go: extracting github.com/google/uuid v1.1.1
go: extracting github.com/mholt/certmagic v0.6.2-0.20190624175158-6a42ef9fe8c2
go: downloading github.com/go-acme/lego v2.5.0+incompatible
go: downloading golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
go: downloading github.com/klauspost/cpuid v1.2.0
go: extracting github.com/klauspost/cpuid v1.2.0
go: extracting golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
go: extracting github.com/go-acme/lego v2.5.0+incompatible
go: downloading golang.org/x/net v0.0.0-20190328230028-74de082e2cca
go: downloading github.com/cenkalti/backoff v2.1.1+incompatible
go: downloading github.com/miekg/dns v1.1.3
go: downloading gopkg.in/square/go-jose.v2 v2.2.2
go: extracting github.com/cenkalti/backoff v2.1.1+incompatible
go: extracting gopkg.in/square/go-jose.v2 v2.2.2
go: extracting golang.org/x/net v0.0.0-20190328230028-74de082e2cca
go: extracting github.com/miekg/dns v1.1.3
go: downloading golang.org/x/text v0.3.0
go: downloading golang.org/x/sys v0.0.0-20190228124157-a34e9553db1e
go: extracting golang.org/x/sys v0.0.0-20190228124157-a34e9553db1e
go: extracting golang.org/x/text v0.3.0
```
```
root@debian:~# cd /root/.caddy/build && go build
go: finding github.com/caddyserver/dnsproviders/gandiv5 latest
go: finding github.com/caddyserver/caddy/caddy/caddymain latest
go: finding github.com/caddyserver/caddy/caddy latest
go: finding github.com/caddyserver/dnsproviders v0.2.0
go: downloading github.com/caddyserver/dnsproviders v0.2.0
go: extracting github.com/caddyserver/dnsproviders v0.2.0
go: finding github.com/exoscale/egoscale v0.17.0
go: finding github.com/aws/aws-sdk-go v1.19.15
go: finding github.com/mholt/caddy v1.0.0-beta2.0.20190420233410-0c3d90ed21a4
go: finding github.com/Azure/go-autorest/autorest/azure/auth v0.1.0
go: finding github.com/Azure/go-autorest/autorest/validation v0.1.0
go: finding github.com/Azure/go-autorest/autorest/to v0.1.0
go: finding github.com/go-ini/ini v1.42.0
go: finding github.com/nrdcg/auroradns v1.0.0
go: finding github.com/Azure/azure-sdk-for-go v27.3.0+incompatible
go: finding gopkg.in/ns1/ns1-go.v2 v2.0.0-20190322154155-0dafb5275fd1
go: finding github.com/smartystreets/goconvey v0.0.0-20190330032615-68dc04aab96a
go: finding github.com/JamesClonk/vultr v2.0.0+incompatible
go: finding google.golang.org/api v0.3.2
go: finding github.com/cloudflare/cloudflare-go v0.8.5
go: finding github.com/google/go-querystring v1.0.0
go: finding github.com/Azure/go-autorest/autorest v0.1.0
go: finding github.com/namedotcom/go v0.0.0-20180403034216-08470befbe04
go: finding github.com/decker502/dnspod-go v0.0.0-20181231011336-d1081970f528
go: finding github.com/json-iterator/go v1.1.6
go: finding github.com/cpu/goacmedns v0.0.1
go: finding github.com/akamai/AkamaiOPEN-edgegrid-golang v0.7.4
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding gopkg.in/ini.v1 v1.42.0
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding go.opencensus.io v0.20.2
go: finding github.com/h2non/gock v0.0.0-00010101000000-000000000000
go: finding github.com/dnsimple/dnsimple-go v0.23.0
go: finding golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: finding github.com/gofrs/uuid v3.2.0+incompatible
go: finding github.com/timewasted/linode v0.0.0-20160829202747-37e84520dcf7
go: finding golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
go: finding github.com/jtolds/gls v4.20.0+incompatible
go: finding github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: finding github.com/dimchansky/utfbom v1.1.0
go: github.com/h2non/gock@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go: finding google.golang.org/grpc v1.19.0
go: finding github.com/Azure/go-autorest/tracing v0.1.0
go: finding github.com/ovh/go-ovh v0.0.0-20181109152953-ba5adb4cf014
go: finding github.com/Azure/go-autorest/autorest/adal v0.1.0
go: finding github.com/Azure/go-autorest/autorest/azure/cli v0.1.0
go: finding github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding golang.org/x/net v0.0.0-20190108225652-1e06a53dbb7e
go: finding golang.org/x/lint v0.0.0-20190301231843-5614ed5bae6f
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/Azure/go-autorest/autorest/mocks v0.1.0
go: finding github.com/Azure/go-autorest/logger v0.1.0
go: finding go.opencensus.io v0.20.1
go: finding honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099
go: finding golang.org/x/sync v0.0.0-20190227155943-e225da77a7e6
go: finding github.com/mholt/certmagic v0.5.0
go: finding golang.org/x/tools v0.0.0-20190312170243-e65039ee4138
go: finding github.com/google/go-cmp v0.2.0
go: finding github.com/juju/ratelimit v1.0.1
go: finding golang.org/x/crypto v0.0.0-20190418165655-df01cb2cc480
go: finding google.golang.org/appengine v1.4.0
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding cloud.google.com/go v0.34.0
go: error loading module requirements
```

As you can see, the build process fails with the error message `go: error loading module requirements` because Go tries to fetch the non-existent repository `github.com/mholt/caddy v1.0.0-beta2.0.20190420233410-0c3d90ed21a4`. In hindsight, this was to be expected because the latest release of this repository (v0.2.0) doesn't include the commit #72 which updated the import paths of all the dnsprovider plugins.

The real problem is that the build process fails even when I specify the latest commit in my `go.mod` file.

```
module caddy

go 1.12

require (
	github.com/caddyserver/caddy v1.0.1
	github.com/caddyserver/dnsproviders 272452b36c2f962599ed59085829141bfc742d87
)
```
```
root@debian:~# cd /root/.caddy/build && go build
go: finding github.com/caddyserver/dnsproviders 272452b36c2f962599ed59085829141bfc742d87
go: finding github.com/dnsimple/dnsimple-go v0.23.0
go: finding github.com/decker502/dnspod-go v0.0.0-20181231011336-d1081970f528
go: finding github.com/nrdcg/auroradns v1.0.0
go: finding github.com/Azure/azure-sdk-for-go v31.0.0+incompatible
go: finding github.com/aws/aws-sdk-go v1.19.15
go: finding github.com/Azure/go-autorest/autorest/azure/auth v0.1.0
go: finding github.com/Azure/go-autorest/autorest/to v0.1.0
go: finding golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: finding google.golang.org/api v0.3.2
go: finding gopkg.in/ns1/ns1-go.v2 v2.0.0-20190322154155-0dafb5275fd1
go: finding github.com/kolo/xmlrpc v0.0.0-20190514182600-74b23a09d7ea
go: finding github.com/ovh/go-ovh v0.0.0-20181109152953-ba5adb4cf014
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190703055746-de62006faf99
go: finding github.com/akamai/AkamaiOPEN-edgegrid-golang v0.7.4
go: finding github.com/Azure/go-autorest/autorest/validation v0.1.0
go: finding google.golang.org/appengine v1.4.0
go: finding github.com/namedotcom/go v0.0.0-20180403034216-08470befbe04
go: finding github.com/google/go-querystring v1.0.0
go: finding github.com/linode/linodego v0.10.0
go: finding github.com/Azure/go-autorest/autorest/adal v0.1.0
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding github.com/Azure/go-autorest/autorest/azure/cli v0.1.0
go: finding github.com/juju/ratelimit v1.0.1
go: finding golang.org/x/crypto v0.0.0-20190418165655-df01cb2cc480
go: finding google.golang.org/grpc v1.19.0
go: finding github.com/timewasted/linode v0.0.0-20160829202747-37e84520dcf7
go: finding golang.org/x/tools v0.0.0-20190312170243-e65039ee4138
go: finding github.com/google/go-cmp v0.2.0
go: finding github.com/nrdcg/goinwx v0.6.0
go: finding github.com/fatih/structs v1.1.0
go: finding golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
go: finding github.com/exoscale/egoscale v0.17.0
go: finding github.com/baiyubin/aliyun-sts-go-sdk v0.0.0-20180326062324-cfa1a18b161f
go: finding github.com/satori/go.uuid v1.2.0
go: finding golang.org/x/net v0.0.0-20190213061140-3a22650c66bd
go: finding golang.org/x/tools v0.0.0-20190226205152-f727befe758c
go: finding github.com/Azure/go-autorest/tracing v0.1.0
go: finding github.com/jmespath/go-jmespath v0.0.0-20180206201540-c2b33e8439af
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/kr/pretty v0.1.0
go: finding github.com/Azure/go-autorest/autorest v0.1.0
go: finding github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742
go: finding go.opencensus.io v0.20.2
go: finding golang.org/x/tools v0.0.0-20190114222345-bf090417da8b
go: finding golang.org/x/net v0.0.0-20190108225652-1e06a53dbb7e
go: finding github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: finding golang.org/x/sync v0.0.0-20181108010431-42b317875d0f
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/go-ini/ini v1.42.0
go: finding github.com/json-iterator/go v1.1.6
go: finding github.com/transip/gotransip v5.8.2+incompatible
go: finding golang.org/x/sys v0.0.0-20190403152447-81d4e9dc473e
go: finding github.com/Azure/go-autorest/autorest/date v0.1.0
go: finding github.com/dnaeon/go-vcr v0.0.0-20180814043457-aafff18a5cc2
go: finding github.com/gofrs/uuid v3.2.0+incompatible
go: finding contrib.go.opencensus.io/exporter/ocagent v0.4.12
go: finding google.golang.org/genproto v0.0.0-20180817151627-c66870c02cf8
go: finding github.com/smartystreets/goconvey v0.0.0-20190330032615-68dc04aab96a
go: finding golang.org/x/sync v0.0.0-20190227155943-e225da77a7e6
go: finding github.com/Azure/go-autorest/autorest/mocks v0.1.0
go: finding go.opencensus.io v0.20.1
go: finding google.golang.org/appengine v1.1.0
go: finding github.com/client9/misspell v0.3.4
go: finding github.com/cloudflare/cloudflare-go v0.8.5
go: finding github.com/kr/text v0.1.0
go: finding github.com/cpu/goacmedns v0.0.1
go: finding github.com/dimchansky/utfbom v1.1.0
go: finding github.com/golang/protobuf v1.3.1
go: finding google.golang.org/api v0.3.1
go: finding golang.org/x/net v0.0.0-20190311183353-d8887717615a
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding golang.org/x/lint v0.0.0-20181026193005-c67002cb31c3
go: finding golang.org/x/exp v0.0.0-20190121172915-509febef88a4
go: finding github.com/json-iterator/go v1.1.5
go: finding github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: finding golang.org/x/sys v0.0.0-20180830151530-49385e6e1522
go: finding google.golang.org/grpc v1.19.1
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/JamesClonk/vultr v2.0.0+incompatible
go: finding gopkg.in/resty.v1 v1.9.1
go: finding github.com/golang/mock v1.1.1
go: finding github.com/kr/pty v1.1.1
go: finding golang.org/x/net v0.0.0-20180724234803-3673e40ba225
go: finding github.com/Azure/go-autorest/logger v0.1.0
go: finding github.com/grpc-ecosystem/grpc-gateway v1.8.5
go: finding cloud.google.com/go v0.34.0
go: finding github.com/apache/thrift v0.12.0
go: finding github.com/h2non/gock v0.0.0-00010101000000-000000000000
go: finding github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
go: finding golang.org/x/net v0.0.0-20180826012351-8a410e7b638d
go: finding github.com/prometheus/client_model v0.0.0-20190115171406-56726106282f
go: finding github.com/hashicorp/golang-lru v0.5.0
go: finding golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
go: finding github.com/jtolds/gls v4.20.0+incompatible
go: finding cloud.google.com/go v0.26.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding golang.org/x/lint v0.0.0-20190301231843-5614ed5bae6f
go: finding golang.org/x/lint v0.0.0-20190227174305-5b3e6a55c961
go: finding gopkg.in/resty.v1 v1.12.0
go: finding golang.org/x/net v0.0.0-20181220203305-927f97764cc3
go: finding github.com/goji/httpauth v0.0.0-20160601135302-2da839ab0f4d
go: github.com/h2non/gock@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go: finding golang.org/x/net v0.0.0-20180611182652-db08ff08e862
go: finding golang.org/x/sys v0.0.0-20181107165924-66b7b1311ac8
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7
go: finding github.com/openzipkin/zipkin-go v0.1.6
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding github.com/ghodss/yaml v1.0.0
go: finding github.com/rogpeppe/fastuuid v0.0.0-20150106093220-6724a57986af
go: finding gopkg.in/ini.v1 v1.42.0
go: finding github.com/prometheus/common v0.2.0
go: error loading module requirements
```

As you can see, Go does not try to import `github.com/mholt/caddy v1.0.0-beta2.0.20190420233410-0c3d90ed21a4` this time around, but it's also not obvious to me what causes the build process to fail now.

@mholt @hairyhenderson Any ideas what needs to be changed in this repo for the build process to finish successfully?

**Comment by mholt** [comment]
Issue duplicated on multiple repos; closing this one in favor of https://github.com/caddyserver/dnsproviders/issues/75.

**Comment by whalehub** [comment]
@mholt I'm confused, this **is** #75.

**Comment by mholt** [comment]
Got my tabs backwards

**Comment by mholt** [comment]
All plugins, including DNS providers, need to use the new import path. (The build server has a hack built in place temporarily to work around this requirement, but I wouldn't wish it upon any individual developer.)

I've just pushed tag v0.3.0, hopefully that will be picked up by go modules.

**Comment by whalehub** [comment]
@mholt Thanks for the new release, but I already tried specifying the latest commit and that didn't solve the issue. I just noticed the line `go: github.com/h2non/gock@v0.0.0-00010101000000-000000000000: unknown revision 000000000000` in the build log and googled the error message, which lead me to https://github.com/h2non/gock/issues/50 and https://github.com/caddyserver/dnsproviders/issues/59. That issue is the reason why the build process still fails. I added `replace github.com/h2non/gock => gopkg.in/h2non/gock.v1 v1.0.14` at the bottom of my `go.mod` file as suggested in those issues and that allowed me to build Caddy successfully just now.
